TrackletAlgorithm/MatchProcessor.h

```diff
@@ -386,40 +998,78 @@ void MatchCalculator(BXType bx,
   ap_int<11> phi_corr      = full_phi_corr >> kPhi_corr_shift;                        // only keep needed bits
+  //ap_uint<3> shifttmp = 6;
+  constexpr auto shifttmp = (LAYER < trklet::N_LAYER) ? log2barrel : log2disk;
+  if(isDisk && isPSStub)
+    phi_corr = (stub_ps_z * proj_phid) >> shifttmp;
+  else if(isDisk && !isPSStub)
+    phi_corr = (stub_2s_z * proj_phid) >> shifttmp;
   ap_int<12> z_corr        = (full_z_corr + (1<<(kZ_corr_shift-1))) >> kZ_corr_shift; // only keep needed bits
+  ap_int<12> r_corr        = full_r_corr >> kr_corr_shift; // only keep needed bits
    
   // Apply the corrections
   const int kProj_phi_len = AllProjection<APTYPE>::kAProjPhiSize + 1;
-  ap_int<kProj_phi_len> proj_phi_corr = proj_phi + phi_corr;  // original proj phi plus phi correction
+  ap_int<kProj_phi_len> proj_phi_corr = proj_phi + phi_corr;  // original proj phi plus phi correction iphi in emulation
   ap_int<13> proj_z_corr   = proj_z + z_corr;      // original proj z plus z correction
 
   // Get phi and z difference between the projection and stub
-  ap_int<10> delta_z         = stub_z - proj_z_corr;
+  ap_int<12> delta_z        = stub_z - proj_z_corr;
   ap_int<14> delta_z_fact   = delta_z * kFact;
-  ap_int<18> stub_phi_long  = stub_phi;         // make longer to allow for shifting
+  const ap_int<18> &stub_phi_long  = stub_phi;         // make longer to allow for shifting
   const ap_int<18> &proj_phi_long  = proj_phi_corr;    // make longer to allow for shifting
-  ap_int<18> shiftstubphi   = stub_phi_long << kPhi0_shift;                        // shift
-  ap_int<18> shiftprojphi   = proj_phi_long << (kShift_phi0bit - 1 + kPhi0_shift); // shift
-  ap_int<17> delta_phi      = shiftstubphi - shiftprojphi;
-  ap_uint<17> abs_delta_phi = iabs<17>( delta_phi );    // absolute value of delta phi
+  ap_int<18> shiftstubphi   = kPhi0_shift > 0 ? stub_phi_long << kPhi0_shift : stub_phi_long;                        // shift
+  if(isDisk && isPSStub)
+    shiftstubphi = kPhi0_shift > 0 ? stub_ps_phi << kPhi0_shift : stub_ps_phi;                        // shift
+  else if(isDisk && !isPSStub) {
+    shiftstubphi = kPhi0_shift > 0 ? stub_2s_phi << kPhi0_shift : stub_2s_phi;                        // shift
+  }
+  ap_int<18> shiftprojphi = proj_phi_long << (kShift_phi0bit - 1 + kPhi0_shift); // shift
+  constexpr int dphibit = 20;
+  ap_int<dphibit> delta_phi      = shiftstubphi - shiftprojphi;
+  ap_uint<3> shiftprojz     = 7;
+  ap_int<7> proj_r_corr    = (stub_z * proj_zd) >> shiftprojz;
+  if(isDisk && isPSStub)
+    proj_r_corr = (stub_ps_z * proj_zd) >> shiftprojz;
+  else if(isDisk)
+    proj_r_corr = (stub_2s_z * proj_zd) >> shiftprojz;
+  const ap_uint<13> &proj_r_long  = proj_z + proj_r_corr;
+  ap_uint<1> shiftr         = 1;
+  ap_int<12> delta_r        = (stub_r >> shiftr) - proj_r_long; // proj_z = RZ
+  ap_uint<12> tmp_stubr = isProjDisk ? LUT_matchcut_rDSS[ap_uint<12>(stub_2s_r)] : LUT_matchcut_rDSS[ap_uint<12>(stub_r)];
```

Comment:
the depth of LUT_matchcut_rDSS is only 10 (one for each strip in each ring). So the conversion here to ap_uint<12> is not correct. I don't understand the second branch of the tertiary operator when this is not a disk projection. Can yo explain?

Reply:
I was seeing issues when using 10 bits. It might be related to all the shifting we have to do. Any variables with `stub_ps` or `stub_2s` are _only_ for the disks, so the second branch of the tertiary operator just swaps back in the original `stub_r` variable.

Reply:
The second branch is using the LUT, but I would have thought it should juste be stub_r?

Reply:
I took another look, and I only use this variable for the 2S disks. I've removed this ternary operator, and renamed the variable to `disk_stubr` to make it clear.